openstack/swift/templates/etc/_proxy-server.conf.tpl

```diff
@@ -23,10 +24,15 @@ log_level = INFO
 {{- end }}
 
 [pipeline:main]
-{{- if eq $context.release "mitaka" }}
+{{- if le $swift_release "M" }}
+# Mitaka pipeline
 pipeline = catch_errors gatekeeper healthcheck proxy-logging cache cname_lookup domain_remap bulk tempurl ratelimit authtoken keystone sysmeta-domain-override staticweb container-quotas account-quotas slo dlo versioned_writes proxy-logging proxy-server
-{{- else }}
+{{- else if eq $swift_release "P" }}
+# Pike pipeline
 pipeline = catch_errors gatekeeper healthcheck proxy-logging cache cname_lookup domain_remap bulk tempurl ratelimit authtoken keystone sysmeta-domain-override staticweb copy container-quotas account-quotas slo dlo versioned_writes proxy-logging proxy-server
+{{- else if ge $swift_release "Q" }}
+# Queens or > pipeline
+pipeline = catch_errors gatekeeper healthcheck proxy-logging cache cname_lookup domain_remap bulk tempurl ratelimit authtoken keystone sysmeta-domain-override staticweb copy container-quotas account-quotas slo dlo versioned_writes symlink proxy-logging proxy-server
```

Comment:
I double-checked the specification for text/template (which says that it uses the comparison rules defined by Go), then the Go language spec, and it turns out you can just compare the release names directly and it will do the right thing. (String comparison is defined to be lexicographic.) See https://play.golang.org/p/lX3F2h3Mo0r

Reply:
Done